fw/cache.c

```diff
@@ -3029,13 +3036,26 @@ cache_req_process_node(TfwHttpReq *req, tfw_http_cache_cb_t action)
 	}
 
 	resp = tfw_cache_build_resp(req, ce, lifetime);
+
+	if (resp && stale) {
+		req->resp = NULL;
+		req->stale_resp = resp;
```

Comment:
I don't agree with this commit. What purpose to send request to server and after receiving response send stale response?? It is very strange for me! If we receive new fresh response we should send it. If this directive is set we should send stale response to client without accessing backend. We should check status of response saved in cache and if it is in  `cache_use_stale` send it without accessing backend. For error cases we can send stale successful responses after receiving error from backend.
For example if I add  
`param(name="200", status=200, use_stale="200", r2_hdrs=[]),` to TestCacheUseStale in your branch. I receive stale 200 response after receiving successful response from server!

Reply:
You are wrong. You got stale response because you specified `cache_use_stale 200` in the config. It's desired behavior, however, maybe we should deny to use status codes except 4x and 5x.

This feature need in case when you for instance got 500 status code, but you have stale response and you reply with stale resp if it configured.

Error will be if you specify `cache_use_stale 500`, but you received 200 stale response.

Reply:
After discussion we decide that administrator can't set 2** or 3** in cache_use_stale

Reply:
Done